homeworks/week02/README.md

```diff
@@ -0,0 +1,232 @@
+```.movie``` and ```.description``` selectors as well.
+
+Try changing the index of ```movieArray``` and reloading to see
+different genres displayed. For example: ```movieArray[3]``` or ```movieArray[6]```.
+
+Commit your code with a helpful comment before proceeding.
+
+## Step 2:
+
+Next, expand on Step 1 so your code renders the entire list of movies
+into HTML at once.
+
+Do this by modifying the ```render()``` function:
+
+1. Add a new string variable
+
+2. Loop over each object in ```movieArray```, and for each one:
+  1. Call your ```renderMovie()``` function, passing in a single movie
+  object
+  2. Concatenate the return value from this function call to your new
+  string variable.
+
+3. Return this new string variable, as your function's return value
+
+If you reload your page, you should now see a list displayed of all
+movies, with whatever styling you have added for each different genre.
+
+Again, commit your code with a helpful comment before proceeding.
+
+## Step 3:
+
+Now we'll demonstrate the power of functions. Without changing your
+```render()``` function, you'll change the way that the list of movies
+gets rendered into HTML. To do this, you'll create a new function that
+creates HTML differently, so the looping code that you just wrote in
+Step 2 won't need to change, except that you'll call this new function
+instead, and you'll see a very different result.
+
+The goal will be to add a new, more compact way of rendering each
+movie, kind of like the "mobile version" of the movie list.
+
+To do this, add a new function that implements the following
+signature:
+
+```javascript
+function renderMovieCompact(movie)
+```
+
+Just like in Step 1, this function should use string concatenation and
+all the attributes of a single object (from the ```movie``` argument)
+to produce a single string of HTML, which the function should then
+return.
+
+The only difference is that now you will create a more compact
+```<li>``` rather than a ```<div>```. So the same movie object from
+Step 1 should now produce this HTML:
+
+```html
+<li class="movie animation">
+  <span class="name">Minions</span>
+  (<span class="year">2015</span>,
+  <span class="genre"> animation</span>)
+  starring: Sandra Bullock, Jon Hamm, Michael Keaton
+</li>
+```
+
+Modify ```render()``` so that instead of calling ```renderMovie()```,
+it calls ```renderMovieCompact()``` instead. If you reload the page,
+you should now see a more compact movie list.
+
+As always, commit your code with a helpful comment before proceeding.
+
+## Step 4
+
+Finally, we'll put everything together.
+
+Modify the body of your loop in ```render()``` by adding an ```if```
+statement, so that if ```displayWide``` is ```true```, you call
+```renderMovie()```, otherwise, call ```renderMovieCompact()```.
+
+Now, if you resize your browser window (to less than 500px) and
+reload, you should see the compact version, but if your browser is
+wider, you should see the full version.
```

Comment:
Good place to define "responsive"? This is jargon in the worst way—it means something completely different in web jargon than in English.